plugins/coverbrowser.koplugin/mosaicmenu.lua

```diff
@@ -731,6 +748,33 @@ function MosaicMenuItem:paintTo(bb, x, y)
         corner_mark:paintTo(bb, x+ix, y+iy)
     end
 
+    -- Paint Progressbar if needed
+    local bookinfo = BookInfoManager:getBookInfo(self.filepath, self.do_cover_image)
+    local percent_finished
+    local status
+    if bookinfo then
+        if self.menu.cover_info_cache[self.filepath] then
+            percent_finished, status = unpack(self.menu.cover_info_cache[self.filepath])
+        else
+            local docinfo = DocSettings:open(self.filepath)
+            -- We can get nb of page in the new 'doc_pages' setting, or from the old 'stats.page'
+            if docinfo.data.summary and docinfo.data.summary.status then
+                status = docinfo.data.summary.status
+            end
+            percent_finished = docinfo.data.percent_finished
+            self.menu.cover_info_cache[self.filepath] = {percent_finished, status}
+        end
+        if status == "complete" then
+            -- Display these instead of the read %
+            percent_finished = nil
+        end
+    end
+    if percent_finished ~= nil then
+        local pos_x = x+self.width - progress_widget.width - ((self.width - self[1][1][1].width) / 2) -progress_widget.bordersize
+        progress_widget.percentage = percent_finished
```

Comment:
`progress_widget:setPercentage(self.percent_finished)`
(Will be exactly the same, but in case one day we do more stuff/reset in this setPercentage, this will be fine.)